src/data/matrix/basic.lean

```diff
@@ -728,15 +728,22 @@ by { ext, apply neg_dot_product }
 lemma mul_vec_neg (v : n → α) (A : matrix m n α) : mul_vec A (-v) = - mul_vec A v :=
 by { ext, apply dot_product_neg }
 
-lemma smul_mul_vec_assoc (A : matrix n n α) (b : n → α) (a : α) :
+lemma smul_mul_vec_assoc (A : matrix m n α) (b : n → α) (a : α) :
```

Comment:
Can you move this up to line 642? I think this lemma only requires `semiring α`, so is in the wrong section.

Reply:
When I move it there, I get an error "(deterministic) timeout"

Reply:
It doesn't do that for me... Maybe just make the change anyway and see if the CI build is ok with it?

Reply:
I got the problem, moving it to line 642 means that the type is a semiring, not a comm_semiring. I don't think that there is the correct place, since comm_semiring is required.

Reply:
> edit : I was moving the wrong lemma